conntrack/cilium_conntrack_linux.go

```diff
@@ -77,8 +76,12 @@ func listRecords(maps []interface{}, clockSource ClockSource, filter EntriesFilt
 			}
 		}
 		if err = m.DumpWithCallback(cb); err != nil {
+			m.Close()
```

Comment:
Consider moving `cb := ` definition outside of for-loop as it does not appear to depend on `m`. And moving whatever remains into a separate function that could retain `defer m.Close()`.